streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -428,7 +428,7 @@ public void shouldCommitActiveAndStandbyTasks() {
         task00.setCommitNeeded();
         task01.setCommitNeeded();
 
-        assertThat(taskManager.commitAll(), equalTo(2));
+        assertThat(taskManager.commitAll(), equalTo(1));
```

Comment:
This is actually an incorrect test -- on commit, we would never do anything with `StandbyTasks` because they always return `false` on `commitNeeded()` -- the used `StateMachineTask` got a corresponding fix.

Reply:
Sounds correct -- I was double checking this code and this issue was introduced in the refactoring via https://github.com/apache/kafka/commit/4090f9a2b0a95e4da127e4786007542276d97520

To not block this PR further, I would suggest to keep this change as-is and do a follow up PR to fix the issue.

Reply:
I will make a separate PR to fix this.